src/index.ts

```diff
@@ -174,7 +174,14 @@ function shouldCommit(details: Element, menu: DetailsMenuElement, event: Event)
 
   if (event.type === 'click') {
     const menuitem = target.closest('[role="menuitem"], [role="menuitemradio"]')
-    const onlyCommitOnChangeEvent = menuitem && menuitem.tagName === 'LABEL' && menuitem.querySelector('input')
+
+    // Ignore double event caused by inputs nested in labels
+    if (menuitem?.tagName === 'LABEL' && target !== menuitem) return
```

Comment:
Instead of checking for `menuItem` each time, could we add an `if (!menuitem) return` right after the `closest` call?